tests/phpunit/test-purger.php

```diff
@@ -1866,4 +1866,30 @@ public function test_filter_post_type() {
 		// Clean up after the test.
 		$this->after_filter_ignore_posts();
 	}
+
+	public function test_future_post_clears_on_published() {
+		$future_date = date('Y-m-d H:i:s', strtotime('+1 day')); // Schedules for 1 day later
+
+		$result = wp_update_post(array(
+			'ID'           => $this->post_id4,
+			'post_status'  => 'future',
+			'post_date'    => $future_date,
+			'post_date_gmt'=> get_gmt_from_date($future_date),
+		), true);
+
+		$this->assertFalse(is_wp_error($result)); # Didn't throw an error updating
+
+		# Intercept the filter added to clear_post_path
```

Comment:
```suggestion
		// Intercept the filter added to clear_post_path
```

Reply:
One of these days I'll figure out how PHP works.